### PR TITLE
chore: Migrate PanelViewModel to TS

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/view_model/PanelViewModel.ts [hunk 1]
```diff
@@ -31,10 +42,44 @@ import {ParticipantDevicesViewModel} from './panel/ParticipantDevicesViewModel';
...
+export type PanelParams = {
+  addMode?: boolean;
+  entity?: User | Message;
+  highlighted?: User[];
+  showLikes?: boolean;
+};
+
 export class PanelViewModel {
+  mainViewModel: MainViewModel;
+  repositories: PanelRepositories;
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Please use `ViewModelRepositories` from `MainViewModel`.

File: src/script/view_model/PanelViewModel.ts [hunk 2]
```diff
@@ -75,15 +120,15 @@ export class PanelViewModel {
...
+    }, {} as Record<string, any>);
   }
 
   /**
    * View model for the details column.
    * @param {MainViewModel} mainViewModel Main view model
    * @param {Object} repositories Object containing all repositories
    */
-  constructor(mainViewModel, repositories) {
+  constructor(mainViewModel: MainViewModel, repositories: PanelRepositories) {
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Please use `ViewModelRepositories` from `MainViewModel`.

File: src/script/view_model/PanelViewModel.ts [hunk 3]
```diff
@@ -31,10 +42,44 @@ import {ParticipantDevicesViewModel} from './panel/ParticipantDevicesViewModel';
...
 import {ContentViewModel} from './ContentViewModel';
+import {MainViewModel} from './MainViewModel';
+import {Conversation} from '../entity/Conversation';
+import {User} from '../entity/User';
+import {Message} from '../entity/message/Message';
+import {BasePanelViewModel} from './panel/BasePanelViewModel';
 
 export const OPEN_CONVERSATION_DETAILS = 'PanelViewModel.OPEN_CONVERSATION_DETAILS';
 
+export type PanelRepositories = {
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Please use `ViewModelRepositories` from `MainViewModel`.

File: src/script/view_model/PanelViewModel.ts [hunk 4]
```diff
@@ -152,50 +198,50 @@ export class PanelViewModel {
    * @param {Object} params params to give to the new view.
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Please remove all JSDocs as they don't include anything helpful.

File: src/script/view_model/PanelViewModel.ts [hunk 5]
```diff
@@ -257,16 +304,15 @@ export class PanelViewModel {
...
     this.state(newPanelState);
 
     const panelStateElementId = this.subViews[newPanelState].getElementId();
     if (panelStateElementId) {
-      return $(`#${panelStateElementId}`).addClass('panel__page--visible');
+      const element = document.querySelector(`#${panelStateElementId}`);
+      element?.classList.add('panel__page--visible');
+      return element;
     }
+    return null;
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Shouldn't it return `undefined` instead?

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: src/script/view_model/PanelViewModel.ts [hunk 6]
```diff
@@ -133,16 +180,15 @@ export class PanelViewModel {
    *
    * @returns {void} nothing returned
    */
-  closePanel() {
+  async closePanel(): Promise<boolean> {
     if (this.isAnimating()) {
       return Promise.resolve(false);
```
Comment (ffflorian) [inline src/script/view_model/PanelViewModel.ts]: Just use `return false`.

File: src/script/view_model/panel/MessageDetailsViewModel.ts [hunk 7]
```diff
@@ -32,6 +32,7 @@ import type {TeamRepository} from '../../team/TeamRepository';
 import type {User} from '../../entity/User';
 import type {ContentMessage} from '../../entity/message/ContentMessage';
 import type {MemberMessage} from '../../entity/message/MemberMessage';
+import {PanelParams} from '../PanelViewModel';
```
Comment (ffflorian) [inline src/script/view_model/panel/MessageDetailsViewModel.ts]: Please use `import type`.

**Review by ffflorian — CHANGES_REQUESTED** [review]

**Review by lipis — APPROVED** [review]

**Review by ffflorian — APPROVED** [review]

**Pull request merged**